naucse/views.py

```diff
@@ -1,119 +1,134 @@
-import calendar
 import datetime
-import logging
-import os
-from urllib.parse import urljoin
 from pathlib import Path
+import functools
+import calendar
+import os
 
+from flask import Flask, render_template, jsonify, url_for, Response, abort, g
```

Comment:
```suggestion
from flask import Flask, render_template, jsonify, url_for, Response, abort, g, redirect
```

The function is used in the runs route, but not imported.

Reply:
Nice catch. I think the redirect there is not very useful and we should remove that call instead. Teh website is frozen, this doesn't work on production.